### ISSUE TITLE
Add terminal configuration in projectorganizer

### ISSUE THREAD

**Issue by RaphaelRochet** [geany/geany-plugins#1383]
Even if  `xterm` and `/usr/bin/x-terminal-emulator` alternative defaults cover a lot of systems, it will fail if you don't use xterm or a debian-compatible distro.

Could it be configurable via plugin preferences, or project properties ?
Or simply use the terminal we can already set in Geany preferences > tools > terminal ?

Thanks

**Comment by techee** [comment]
@RaphaelRochet Done in #1432. Would you give it a try?

**Comment by techee** [comment]
Well, I think the only option is to install Geany and geany-plugins from sources. Geany provides nightly builds for Debian only and as the pull request isn't merged yet, nightly builds will be unavailable anyway.

**Comment by RaphaelRochet** [comment]
Sorry, couldn't compile it. Took a long time to gather dependencies and decrypt autogen nonsense errors, but hit a dead end ...
The configure script looks broken.
```
/home/raphael/geany-plugins/configure: line 6447: syntax error near unexpected token `newline'
/home/raphael/geany-plugins/configure: line 6447: `yes:'
```

**Comment by techee** [comment]
You are talking to a guy who does his best not to understand autotools to preserve his sanity ;-)

Anyway, if it's line 17, have you installed libtool? Also, have you successfully compiled and installed Geany too? geany-plugins will require the not yet released 2.1 version.

**Comment by RaphaelRochet** [comment]
Yeah that autotools thing looks crazy ... no clue what it's complaining about :(
I really made my best,  libtool 2.5.4.23-5b58-dirty and geany 2.1 (git >= 876242d8b) .

**Comment by techee** [comment]
What exactly is on line 6447 where this error happens?

**Comment by RaphaelRochet** [comment]
6447 is the "yes:" line. Autogen outputs `is m4_require'd but not m4_defun'd` just after that part.  But had no luck finding out what that means :(

I guess some chunk of code is missing here, causing that syntax error.
```
...
  if test "x$ac_ct_CC" = x; then
    CC=""
  else
    case $cross_compiling:$ac_tool_warned in
yes:

{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: using cross tools not prefixed with host triplet" >&5
printf "%s\n" "$as_me: WARNING: using cross tools not prefixed with host triplet" >&2;}
ac_tool_warned=yes ;;
esac
    CC=$ac_ct_CC
  fi
else
  CC="$ac_cv_prog_CC"
fi
...
```

**Comment by techee** [comment]
Interesting. The corresponding piece of code looks this way on my machine:

```bash
  if test "x$ac_ct_CC" = x; then
    CC=""
  else
    case $cross_compiling:$ac_tool_warned in
yes:)
{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: using cross tools not prefixed with host triplet" >&5
printf "%s\n" "$as_me: WARNING: using cross tools not prefixed with host triplet" >&2;}
ac_tool_warned=yes ;;
esac
    CC=$ac_ct_CC
  fi
else
  CC="$ac_cv_prog_CC"
fi

fi
```

The only differences I see is `)` after `yes:` and no empty line after it. What happens if you modify it this way (stupid workaround but I have no idea why your output differs).

**Comment by RaphaelRochet** [comment]
Did it, unfortunately it fails later : 
```
./configure: line 7293: syntax error near unexpected token `)'
./configure: line 7293: `)'
```
Line 7293 is the lonely `)` here ... could it be the `)` that was missing earlier .. man I feel my sanity is fading away quickly ...
```
...
 if
  test "x$enable_dependency_tracking" != xno \
  && test "$am_cv_CC_dependencies_compiler_type" = gcc3; then
  am__fastdepCC_TRUE=
  am__fastdepCC_FALSE='#'
else
  am__fastdepCC_TRUE='#'
  am__fastdepCC_FALSE=
fi

)

# Check whether --enable-static was given.
...
```

Will try to remove it. What can go wrong ...

**Comment by RaphaelRochet** [comment]
Well, that worked.

I could try your plugin preferences, worked like a charm. 👍 
Tanks for your work and your patience 🥇

**Comment by techee** [comment]
Good to hear! Miracles sometimes happen :-)

I'm going to merge the PR and close this issue. Thanks for testing!

**Comment by b4n** [comment]
Re the autotools issue, it probably stems from an earlier error when calling autogen, we'd need the log for that. But that should likely be another issue if any.
